File: app/src/main/java/com/wealoha/social/beans/instagram/OauthService.java
package com.wealoha.social.beans.instagram;

import retrofit.Callback;
import retrofit.http.Field;
import retrofit.http.FormUrlEncoded;
import retrofit.http.POST;

public interface OauthService {

	@POST("/oauth/access_token")
	@FormUrlEncoded
	public AccessToken accessToken(@Field("client_id") String clientId, //
			@Field("client_secret") String clientSecret, //
			@Field("grant_type") String grant_type, //
			@Field("redirect_uri") String redirectUri, //
			@Field("code") String code);

	@POST("/oauth/access_token")
	@FormUrlEncoded
	public void accessToken(@Field("client_id") String clientId, //
			@Field("client_secret") String clientSecret, //
			@Field("grant_type") String grant_type, //
			@Field("redirect_uri") String redirectUri, //
			@Field("code") String code,//
			Callback<AccessToken> callback);
}

File: app/src/main/java/com/wealoha/social/beans/instagram/OauthService.kt
package com.wealoha.social.beans.instagram

import retrofit.Callback
import retrofit.http.Field
import retrofit.http.FormUrlEncoded
import retrofit.http.POST

interface OauthService {
    @POST("/oauth/access_token")
    @FormUrlEncoded
    fun accessToken(
        @Field("client_id") clientId: String?,  //
        @Field("client_secret") clientSecret: String?,  //
        @Field("grant_type") grant_type: String?,  //
        @Field("redirect_uri") redirectUri: String?,  //
        @Field("code") code: String?
    ): AccessToken?

    @POST("/oauth/access_token")
    @FormUrlEncoded
    fun accessToken(
        @Field("client_id") clientId: String?,  //
        @Field("client_secret") clientSecret: String?,  //
        @Field("grant_type") grant_type: String?,  //
        @Field("redirect_uri") redirectUri: String?,  //
        @Field("code") code: String?,  //
        callback: Callback<AccessToken?>?
    )
}